notebooks/custom-prompts-demo.ipynb

```diff
@@ -0,0 +1,571 @@
+    "            response_num = int(summary[start+1:end-1])\n",
+    "            doc_num = self.responses[response_num-1]['documentIndex']\n",
+    "            metadata = {item['name']: item['value'] for item in self.docs[doc_num]['metadata']}\n",
+    "            text = extract_between_tags(self.responses[response_num-1]['text'], start_tag, end_tag)\n",
+    "            if 'url' in metadata.keys():\n",
+    "                url = f\"{metadata['url']}#:~:text={quote(text)}\"\n",
+    "                if url not in self.refs:\n",
+    "                    self.refs.append(url)\n",
+    "\n",
+    "        # replace references with markdown links\n",
+    "        refs_dict = {url:(inx+1) for inx,url in enumerate(self.refs)}\n",
+    "        for match in reversed(matches):\n",
+    "            start, end = match\n",
+    "            response_num = int(summary[start+1:end-1])\n",
+    "            doc_num = self.responses[response_num-1]['documentIndex']\n",
+    "            metadata = {item['name']: item['value'] for item in self.docs[doc_num]['metadata']}\n",
+    "            text = extract_between_tags(self.responses[response_num-1]['text'], start_tag, end_tag)\n",
+    "            if 'url' in metadata.keys():\n",
+    "                url = f\"{metadata['url']}#:~:text={quote(text)}\"\n",
+    "                citation_inx = refs_dict[url]\n",
+    "                summary = summary[:start] + f'[\\[{citation_inx}\\]]({url})' + summary[end:]\n",
+    "            else:\n",
+    "                summary = summary[:start] + summary[end:]\n",
+    "\n",
+    "        return summary"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ba703011-f594-4c71-8c6f-42aaa9ff281b",
+   "metadata": {},
+   "source": [
+    "The VectaraQuery class simplifies running a query against a Vectara corpus.\n",
+    "Note that in addition to the expected customer_id, corpus_id and API key, it has two additional arguments in the constructor: `prompt_name` and `prompt_text`, which are the name of the prompt we would use and the text of the prompt when we use custom prompts. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "63fda689-6fa2-4543-aa45-fed5942cbab6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class VectaraQuery():\n",
+    "    def __init__(self, api_key: str, customer_id: str, corpus_id: str, prompt_name: str = None, prompt_text: str = None):\n",
+    "        self.customer_id = customer_id\n",
+    "        self.corpus_id = corpus_id\n",
+    "        self.api_key = api_key\n",
+    "        self.prompt_name = prompt_name if prompt_name else \"vectara-summary-ext-v1.2.0\"\n",
+    "        self.prompt_text = prompt_text\n",
+    "\n",
+    "    def get_body(self, query_str: str):\n",
+    "        corpora_key_list = [{\n",
+    "                'customer_id': self.customer_id, 'corpus_id': self.corpus_id, 'lexical_interpolation_config': {'lambda': 0.0}\n",
```

Comment:
Is there a specific reason you're setting lambda here to 0.0?

Reply:
Not really. Let me change back to default.